### PR TITLE
DDF-5426 Ensures the "not found" route is the last route

### PR DESCRIPTION
**Pull request by stevenmalmgren**
#### What does this PR do?
Moves the not found route to the last position to ensure other routes aren't overwritten - for example if the routes are extended. 
#### Who is reviewing it? 

@andrewkfiedler 
@Bdthomson 
@djblue 
@vinamartin 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.

@andrewkfiedler
@vinamartin 

#### How should this be tested?
This can be tested by navigating to different routes and ensuring you are directed where you expected. 

#### Any background context you want to provide?
If you wanted to override routes down stream the required order would cause the not found route to be before your extended routes. 
#### What are the relevant tickets?
Fixes: #05426

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by jrnorth** [comment]
@stevenmalmgren Issue number?

**Comment by Bdthomson** [comment]
build now

**Comment by jordanwilking** [comment]
💅   Hero Successful  💅 

#### Steps

1.  Pulled and ran a quick build
1.  Clicked down all of the links in the left nav bar

File: ui/packages/catalog-ui-search/src/main/webapp/js/router.js [hunk 1]
```diff
@@ -25,20 +25,20 @@ import React from 'react'
 import { render } from 'react-dom'
 import ExtensionPoints from '../extension-points'
 
-const { routes } = ExtensionPoints
-
-// notfound route needs to come at the end otherwise no other routes will work
-render(<ReactRouter routeDefinitions={routes} />, Application.App.router.$el[0])
+render(
+  <ReactRouter routeDefinitions={ExtensionPoints.routes} />,
+  Application.App.router.$el[0]
```
Comment (cantstoptheunk) [inline ui/packages/catalog-ui-search/src/main/webapp/js/router.js]: ❓ What does this exactly grab?

Comment (jordanwilking) [reply ui/packages/catalog-ui-search/src/main/webapp/js/router.js]: That was already there. It was just reformatted.

**Pull request merged**